README.md

```diff
@@ -1,407 +1,436 @@
+| Whether to use less or bat .. TODO                    | `use_less`             | boolean                    |
 
-What this does:
+#### Sorting
 
-* Make the paths full size by default. On LSP references we are shortening paths.
-* Bind `<ctrl-p>` for a common mapping to find files.
-  - Using `telescope.builtin.git_files` is better in git directories. You can make a toggle to detect if it's a git directory.
-* Bind `gr` to find references in LSP.
-  - `telescope.builtin.lsp_workspace_symbols` and `telescope.builtin.lsp_document_symbols` are also good to bind for LSP.
+| Description                                           | Keys                   | Options                    |
+|-------------------------------------------------------|------------------------|----------------------------|
+| The sorter for file lists.                            | `file_sorter`          | [see sorters](#built-in-sorters)    |
+| The sorter for everything else.                       | `generic_sorter`       | [see sorters](#built-in-sorters)    |
+| The command line argument for grep search ... TODO.   | `vimgrep_arguments`    | dict                               |
+| ... TODO                                              | `selection_strategy`   | follow/reset/row                   | 
+| Pattern to be ignored `{ "scratch/.*", "%.env"}`      | `file_ignore_patterns` | dict                               | 
+| Whether to shorten paths or not.                      | `shorten_path`         | boolean                            |
 
-## Mappings
 
-Mappings are fully customizable. Many familiar mapping patterns are setup as defaults.
+#### Defaults
 
-```
-<C-n>  <C-p> next | previous
-<Down> <Up>  next | previous
-j      k     next | previous (in normal mode)
-<CR>         go to file selection
-
-<C-x>        go to file selection as a split
-<C-v>        go to file selection as a vertical split
-<C-t>        go to a file in a new tab
-
-<C-u>        scroll up in preview window
-<C-d>        scroll down in preview window
-
-<C-c>        close telescope
-<Esc>        close telescope (in normal mode)
-```
-
-To see the full list of mappings, check out `lua/telescope/mappings.lua` and the `default_mappings` table.
-
-To override ALL of the default mappings, you can use the `default_mappings` key in the `setup` table.
+As an example of using the `setup()` method,
+  the following code configures `telescope.nvim`
+  to its default settings.
 
+```lua
+require('telescope').setup{
+  defaults = {
+    vimgrep_arguments = {
+      'rg', 
+      '--color=never', 
+      '--no-heading', 
+      '--with-filename', 
+      '--line-number', 
+      '--column', 
+      '--smart-case'
+    },
+    prompt_position = "bottom",
+    prompt_prefix = ">",
+    selection_strategy = "reset",
+    sorting_strategy = "descending",
+    layout_strategy = "horizontal",
+    layout_defualts = {
```

Comment:
Is this a typo of the word defaults? (layout_defualts -> layout_defaults)

Reply:
Nice catch, I will fix it soon, thanks @PanagiotisS